components/gitpod-db/src/typeorm/migration/1662040283793-AddUsageTable.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class AddUsageTable1662040283793 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(
+            `CREATE TABLE \`d_b_usage\` (
+                \`id\` char(36) NOT NULL,
+                \`attributionId\` varchar(255) NOT NULL,
+                \`description\` varchar(255) NOT NULL,
+                \`creditCents\` bigint NOT NULL,
+                \`effectiveTime\` varchar(255) NOT NULL,
+                \`creationTime\` varchar(255) NOT NULL,
```

Comment:
Should this have an AutoSet on create property?

Reply:
We don't usually do this for such columns. Also, it might be good to use the same time for all the entries we add in one reconciliation run.

Reply:
For me this was intended to be a sort of record metadata. Basically storing two timestamps - First creation, and Last Update (and automatically maintaining this by the DB) would give us an audit when something did change or when it first entered the DB.

For our custom application logic, I'd use the `effectiveTime` but I'm open on this.

Reply:
Ok, I was leaning to make it similar to some of the `creationTime` fields we already have. But I wasn't too sure about it's usefulness and semantic anyway. I'll change it to `_created timestamp(6) ...`.